tests/hpc/slurm_master_backup.pm

```diff
@@ -0,0 +1,72 @@
+# SUSE's openQA tests
+#
+# Copyright © 2017-2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: slurm cluster initialization
+# Maintainer: Sebastian Chlad <sebastian.chlad@suse.com>
+
+use base "hpcbase";
+use strict;
+use warnings;
+use testapi;
+use lockapi;
+use utils;
+
+sub run {
+    my $self  = shift;
+    my $nodes = get_required_var("CLUSTER_NODES");
+
+    $self->prepare_user_and_group();
+    zypper_call('in nfs-client rpcbind');
+
+    systemctl 'start nfs';
+    systemctl 'start rpcbind';
+
+    assert_script_run("showmount -e 10.0.2.1");
+
+    assert_script_run("mkdir -p /shared/slurm");
+    assert_script_run("chown -Rcv slurm:slurm /shared/slurm");
+    assert_script_run("mount -t nfs -o nfsvers=3 10.0.2.1:/nfs/shared /shared/slurm");
+
+    zypper_call('in slurm slurm-munge');
+
+    barrier_wait("SLURM_SETUP_DONE");
+    barrier_wait("SLURM_MASTER_SERVICE_ENABLED");
+
+    # enable and start munge
+    $self->enable_and_start('munge');
+
+    # enable and start slurmctld
+    $self->enable_and_start('slurmctld');
+    systemctl 'status slurmctld';
+
+    # enable and start slurmd since maester also acts as Node here
+    $self->enable_and_start('slurmd');
+    systemctl 'status slurmd';
+
```

Comment:
Looks like repeating pattern, maybe status check could be included in `enable_and_start` or new function could be created `enable_start_check`, but it is not important.

Reply:
@czerw yeah, fully agree. There is quite a few places which could go into certain functions. I will do that and perhaps eventually have two run() - init cluster and then another one for executing the test procedures itself. Further enhancements as you propose will be done with slurm with database test case. And then even further enhancements should be done once I look into other components of the hpc, so eventually hpclib should provide generic functions for preparing fully functional cluster and there should be no duplication of the code in the test procedures.